addons/account/models/account_move.py

```diff
@@ -1307,6 +1307,9 @@ def create(self, vals):
                         'analytic_account_id': vals.get('analytic_account_id') if tax.analytic else False,
                     }
                     bank = self.env["account.bank.statement"].browse(vals.get('statement_id'))
+                    if not bank:
+                        st_line_id = vals.get('statement_line_id')
+                        bank = self.env["account.bank.statement.line"].browse(st_line_id).statement_id
```

Comment:
Because `statement_id` is `related='statement_line_id.statement_id'`, I think you can inline this:
`bank = self.env["account.bank.statement.line"].browse(vals.get('statement_line_id')).statement_id`
instead of the 4 lines above